roles/upgrade/tasks/leapp-upgrade.yml

```diff
@@ -29,6 +29,7 @@
 
 - name: leapp-upgrade | Include update-and-reboot.yml
   ansible.builtin.include_tasks: update-and-reboot.yml
+  when: skip_pre_upgrade_update_reboot is not defined or not skip_pre_upgrade_update_reboot|bool
```

Comment:
When would the var not be defined?

Reply:
Seems reasonable to me otherwise.  I would want to remove the check for defined since it's in the defaults.

Reply:
Hi @jeffmcutter 

Thanks for the feedback.  You're right that it's in the defaults so the check should not be needed.  Unfortunately, in the past I've known users to update individual task files from git rather than pulling all the updates (yes, that's bad and not logical...but that's how I've seen it done).  So someone just pulling the updated task file wouldn't have `skip_pre_upgrade_update_reboot` defined.  I agree that's bad practice, so I'll follow your advice and remove that check.

Before I do, there is another option:

Rather than defining `skip_pre_upgrade_update_reboot` I could write the logic the other way around and instead have something like`pre_upgrade_update` with a default of true.  Right now you have `post_upgrade_update` so maybe a `pre_upgrade_update` fits better?

Happy with either option as both achieves the same result.

Reply:
Hi @rjo-uk,

That sounds good.  Consistent is always good, plus I think it will make the logic more straight forward.

Thanks,
-Jeff